sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -709,7 +715,8 @@ export class MessageReceiver extends LinkEntity {
   async detached(receiverError?: AmqpError | Error): Promise<void> {
     const connectionId = this._context.namespace.connectionId;
     try {
-      const wasCloseInitiated = this._receiver && this._receiver.isItselfClosed();
+      const wasCloseInitiated =
+        this.wasCloseInitiated || (this._receiver && this._receiver.isItselfClosed());
```

Comment:
As per https://github.com/Azure/azure-sdk-for-js/pull/2442#issuecomment-486458024, this change is not needed

Reply:
This would help pre-empt the detached call early on. Would keep it if there is no issue

Reply:
Removed, can discuss offline and add in later if it makes sense.